ui/infra-compare/InfraCompareTableView.jsx

```diff
@@ -247,9 +247,10 @@ InfraCompareTableView.propTypes = {
   }),
   getDisplayResults: PropTypes.func.isRequired,
   getQueryParams: PropTypes.func.isRequired,
-  compareData: PropTypes.shape({
-    size: PropTypes.number,
-  }),
+  compareData: PropTypes.oneOfType([
+    PropTypes.instanceOf(Map),
+    PropTypes.array,
+  ]),
```

Comment:
I was still having issues using `compareData: PropTypes.instanceOf(Map)` as explained below so i went with `oneOfType`.

When I use `compareData: PropTypes.array`, I have this warning:

![image](https://user-images.githubusercontent.com/36057474/85343495-e245f480-b4e4-11ea-866a-2026a6fd8493.png)


And when I use `compareData: PropTypes.instanceOf(Map)`, I have this warning:

![image](https://user-images.githubusercontent.com/36057474/85343552-0dc8df00-b4e5-11ea-8370-e6f1cb45979d.png)

Reply:
Yeah, using `oneOfType` sounds like the right solution.